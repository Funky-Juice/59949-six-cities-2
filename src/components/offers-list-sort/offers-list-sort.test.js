import renderer from 'react-test-renderer';
import OffersListSort from './offers-list-sort';
import offerTestObj from '../../mocks/test-offer';

it(`OffersListSort correctly renders`, () => {
  const answerCB = jest.fn();
  const tree = renderer
    .create(<OffersListSort
      isVisible={false}
      activeCity={offerTestObj.city}
      activeItem={-1}
      activeOffers={[offerTestObj]}
      onItemClick={answerCB}
      onVisibleChange={answerCB}
      setActiveOffers={answerCB}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
