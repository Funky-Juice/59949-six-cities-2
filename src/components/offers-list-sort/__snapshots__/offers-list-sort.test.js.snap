// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OffersListSort correctly renders 1`] = `
<form
  action="#"
  className="places__sorting"
  method="get"
>
  <span
    className="places__sorting-caption"
  >
    Sort by
  </span>
  <span
    className="places__sorting-type"
    onClick={[Function]}
    style={
      Object {
        "paddingLeft": "5px",
      }
    }
  >
    Popular
    <svg
      className="places__sorting-arrow"
      height="4"
      style={
        Object {
          "transform": "rotateX(0deg)",
        }
      }
      width="7"
    >
      <use
        xlinkHref="#icon-arrow-select"
      />
    </svg>
  </span>
  <ul
    className="places__options places__options--custom places__options--opened"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <li
      className="places__option places__option--active"
      onClick={[Function]}
    >
      Popular
    </li>
    <li
      className="places__option false"
      onClick={[Function]}
    >
      Price: low to high
    </li>
    <li
      className="places__option false"
      onClick={[Function]}
    >
      Price: high to low
    </li>
    <li
      className="places__option false"
      onClick={[Function]}
    >
      Top rated first
    </li>
  </ul>
</form>
`;
